topostats/grainstats.py

```diff
@@ -954,3 +959,163 @@ def get_cropped_region(self, image: np.ndarray, length: int, centre: np.ndarray)
             centre[0] - length - shiftx : centre[0] + length + 1 - shiftx,
             centre[1] - length - shifty : centre[1] + length + 1 - shifty,
         ]
+
+    @staticmethod
+    def get_triangle_height(base_point_1: np.array, base_point_2: np.array, top_point: np.array) -> float:
+        """Returns the height of a triangle defined by the input point vectors.
+        Parameters
+        ----------
+        base_point_1: np.ndarray
+            a base point of the triangle, eg: [5, 3].
+
+        base_point_2: np.ndarray
+            a base point of the triangle, eg: [8, 3].
+
+        top_point: np.ndarray
+            the top point of the triangle, defining the height from the line between the two base points, eg: [6,10].
+
+        Returns
+        -------
+        Float
+            The height of the triangle - ie the shortest distance between the top point and the line between the two base points.
+        """
+
+        # Height of triangle = A/b = ||AB X AC|| / ||AB||
+        a_b = base_point_1 - base_point_2
+        a_c = base_point_1 - top_point
+        height = np.linalg.norm(np.cross(a_b, a_c)) / np.linalg.norm(a_b)
+        return height
+
+    @staticmethod
+    def get_max_min_ferets(edge_points: list):
+        """Returns the minimum and maximum feret diameters for a grain.
+        TODO: Add more description
+        """
+
+        # plt.figure(figsize=(10,10))
+
+        # Sort the vectors by their x coordinate and then by their y coordinate.
+        # The conversion between list and numpy array can be removed, though it would be harder
+        # to read.
+        # print(f'edge_points: {edge_points}')
+        edge_points.sort()
+        edge_points = np.array(edge_points)
```

Comment:
I agree, I tried using numpy arrays however it got bulky since it just so happens that the default sorting functionality is precisely what is needed, and my attempts to replicate it using numpy sorts always had some niche cases where it would not work as expected, mostly due to repeated values in x or y. I did manage to get a numpy version working, though it was quite a bit bulkier than the standard python version so I just kept it simple for now.